applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -447,11 +363,21 @@ static void stream_recv_cb(struct bt_audio_stream *stream, const struct bt_iso_r
 		bad_frame = true;
 	}
 
-	receive_cb(buf->data, buf->len, bad_frame, info->ts);
+	ret = channel_index_get(stream->conn, &channel_index);
+	if (ret) {
+		LOG_ERR("Channel index not found");
+		return;
+	}
 
-	recv_cnt++;
-	if ((recv_cnt % 1000U) == 0U) {
-		LOG_DBG("Received %d total ISO packets", recv_cnt);
+	if (channel_index == AUDIO_CH_L) {
+		receive_cb(buf->data, buf->len, bad_frame, info->ts);
+	} else if (channel_index == AUDIO_CH_R) {
+		packet_count++;
```

Comment:
So, if we get right data, the data will just be discarded here and printed. Is it better to send all data to the `receive_cb` and let the higher layer decide what to do?

Reply:
Moved throwing of right channel data to streamctrl.c